### DISCUSSION TITLE
getting error "ValueError: math domain error"

### DISCUSSION THREAD

**Discussion by razieh8051** [Nixtla/statsforecast#654] [Q&A]
I am getting an error about the log function false input when using the` sf.fit(df)` for AutoARIMA. I checked my inputs of 'unique_id', 'ds', 'y' for df which are in correct format:
**unique_id**             int64
**ds**           datetime64[ns]
**y**                   float64
 and there is no NaN or negative numbers that could cause this. Any ideas what could be wrong?

```
File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\statsforecast\arima.py:1487, in Arima(x, order, seasonal, xreg, include_mean, include_drift, include_constant, blambda, biasadj, method, model, **kwargs)
   1485 nstar = n - tmp["arma"][5] - tmp["arma"][6] * tmp["arma"][4]
   1486 tmp["aicc"] = tmp["aic"] + 2 * npar * (nstar / (nstar - npar - 1) - 1)
-> 1487 tmp["bic"] = tmp["aic"] + npar * (math.log(nstar) - 2)
   1488 tmp["xreg"] = xreg
   1489 tmp["lambda"] = blambda

ValueError: math domain error
```

**Update:**
Here, in  `fm[i, i_model] = new_model.fit(y=y, X=X)` of [this](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/core.py#L71) line, `y` is supposed to be `(n,)` or `(1,)`? I figured in my input going into the loop it is `(1,)` and probably this what makes `n` negative in [this line](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/arima.py#L1214) and then this `math.log`  [line](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/arima.py#L1488) cannot proceed. 

`ga` in [this line](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/core.py#L874) has the shape as `(n, 1) <class 'numpy.recarray'>`  in both my input and your [sample](https://datasets-nixtla.s3.amazonaws.com/air-passengers.csv) but when enters the loop `for i, grp in enumerate(self):` in the sample provided `grp` becomes `(n, )` but in my input it becomes `(1,)`; not sure why this is the case.

**Comment by tirthshah17** [comment]
Hello @razieh8051! I am getting similar error while running the fit method of statsforecast library. I checked my data and it does have any null values nor any negative values. Did you get any solution to this? If yes, then can you please share it?

**Comment by jmoralez** [comment]
Hey @razieh8051, sorry for the late reply. I've investigated this and there's a bug that only raises an error when there's a single sample, but it's wrong in general. I've opened #690 to track it. In the meantime you can define a fallback model in the statsforecast constructor that will be used when this error happens with arima or you can also drop series with only a single point.

Please let us know if you have further doubts.
